test/spec/authn/mfa-challenge.js

```diff
@@ -1522,6 +1548,162 @@ describe('MFA_CHALLENGE', function () {
         expect(err.errorCauses).toBeUndefined();
       }
     });
+
+    // OKTA-823470: iOS18 polling issue
+    // NOTE: only run these tests in browser environments
+    (!!global.document ? describe : describe.skip)('iOS18 polling', () => {
+      const togglePageVisibility = () => {
+        document.hidden = !document.hidden;
+        document.dispatchEvent(new Event('visibilitychange'));
+      };
+
+      // see https://stackoverflow.com/a/52196951 for more info about jest/promises/timers
+      const advanceTestTimers = async () => {
+        jest.runOnlyPendingTimers();
+        // flushes promise queue
+        await Promise.resolve();
+        await Promise.resolve();
```

Comment:
Can the multiple explicit calls to `Promise.resolve` be replaced with this solution in the SO link you have in the comment? https://stackoverflow.com/a/58716087